hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/result/DefaultErrorHandler.java

```diff
@@ -47,15 +50,17 @@ public void handleDiscarded(Message message, Subscription subscription, MessageS
         updateMeters(subscription);
         updateMetrics(Counters.DISCARDED, message, subscription);
 
-        addToMessageLog(message, subscription, result);
+        MessageMetadata messageMetadata = toMessageMetadata(message, subscription);
+
+        addToMessageLog(message, subscription, result, messageMetadata.getExtraRequestHeaders());
 
         trackers.get(subscription).logDiscarded(toMessageMetadata(message, subscription), result.getRootCause());
     }
 
-    private void addToMessageLog(Message message, Subscription subscription, MessageSendingResult result) {
+    private void addToMessageLog(Message message, Subscription subscription, MessageSendingResult result, Map<String, String> extraRequestHeaders) {
         result.getLogInfo().forEach(logInfo ->
                 undeliveredMessageLog.add(createUndeliveredMessage(subscription, new String(message.getData()), logInfo.getFailure(), clock.millis(),
```

Comment:
Can we use here some kind of builder ? Maybe it's worth to create an `SentMessageTraceBuilder` and use it here. The more the number of arguments grows, the code is less readable

Reply:
It's a good idea. I added a builder class and replaced calls to the constructor with it.